app/services/resource_compare_service.rb

```diff
@@ -0,0 +1,218 @@
+# frozen_string_literal: true
+
+# Compares two PDCMetadata::Resource objects and provides a hash with the `differences`
+#
+# If there are no differences between the two objects `identical?` returns `true` and `differences == {}`
+# If there are differences the `differences` hash has the following structure:
+#
+# ```
+#    :field_name = {action:, from:, to:, value: }     for single-value fields (e.g. description)
```

Comment:
I was wondering if it might make life simpler to just return an array no matter what. This is just a pondering based on having to do a if above.

Reply:
This was a great insight...I ended changing the code to always return an array and it looks much better and simpler.